xds/internal/balancer/clusterimpl/clusterimpl.go

```diff
@@ -247,8 +247,22 @@ func (b *clusterImplBalancer) UpdateClientConnState(s balancer.ClientConnState)
 	}
 
 	if b.config == nil || b.config.ChildPolicy.Name != newConfig.ChildPolicy.Name {
-		if err := b.child.SwitchTo(bb); err != nil {
-			return fmt.Errorf("error switching to child of type %q: %v", newConfig.ChildPolicy.Name, err)
+		if b.config == nil || b.config.ChildPolicy.Name != newConfig.ChildPolicy.Name {
+			strCfg, err := newConfig.ChildPolicy.MarshalJSON()
+			if err != nil {
+				return fmt.Errorf("error marshaling config: %v", err)
+			}
+			r := json.RawMessage(strCfg)
+			sc, err := gracefulswitch.ParseConfig(r)
+			if err != nil {
+				return fmt.Errorf("error parsing child config: %v", err)
+			}
```

Comment:
What we want is something like this where we the JSON configuration passed to the gracefulswitch balancer is an array of LB policy configurations (but with a single entry, one corresponding to the child policy that we have).

```golang
       // Build config for the gracefulswitch balancer. It is safe to ignore JSON
       // marshaling errors here, since the config was already validated as part of
       // ParseConfig().
       cfg := []map[string]any{{newConfig.ChildPolicy.Name: newConfig.ChildPolicy.Config}}
       cfgJSON, _ := json.Marshal(cfg)
       parsedCfg, err := gracefulswitch.ParseConfig(cfgJSON)
       if err != nil {
               return err
        }
```

Reply:
Updated

Reply:
I'm not sure if it produced an array.